src/utils/hooks/use-generate-pathname.js

```diff
@@ -1,12 +1,31 @@
 import { useCallback } from 'react';
 import { useHistory, generatePath } from 'react-router-dom';
-import { localStorageName, routes } from '../../config';
+import {
+  localStorageName,
+  params,
+  routes,
+  queryParamsToRetain,
+} from '../../config';
 
 const getCurrentActivePipeline = () => {
   const localStorage = window.localStorage.getItem(localStorageName);
   return JSON.parse(localStorage)?.pipeline?.active;
 };
 
+/**
+ * Retains only the specified query parameters and removes all others from the given searchParams object.
+ * @param {URLSearchParams} searchParams - The searchParams object to modify.
+ */
+const retainOtherQueryParams = (searchParams) => {
```

Comment:
[Nit] if I understand correctly, this function is used to retain the default queryParams and remove others, can we mention that in the comments (on line 16) and rename this to retainDefaultQueryParams ?

Reply:
Updated.